src/main/drivers/bus_spi_stdperiph.c

```diff
@@ -113,6 +113,8 @@ uint8_t spiTransferByte(SPI_TypeDef *instance, uint8_t txByte)
 {
     uint16_t spiTimeout = 1000;
 
+    instance->DR;
```

Comment:
At least wrap it in `UNUSED()` to state the intent, like done in HAL/LL/SPL code, as well as BF own code.

Reply:
Never thought of using `UNUSED` in this situation; thought it was only for parameters.

May be casting with `(void)`?

Reply:
Have a look at uart flag clearing code in SPL, it uses UNUSED. And UNUSED is usually implemented as a cast to void.

Reply:
Where is that? (Yes, `UNUSED(x)` is `(void)(x)`, but the case here is not unused like in `UNUSED(some_parameter)`, but to "discard".)

Reply:
You don't want to discard anything, if compiler "discards" your read you'll not achieve what you wanted to achieve.

Just a few examples among many:
```
/** @brief  Clear the SPI MODF pending flag.
  * @param  __HANDLE__: specifies the SPI Handle.
  *         This parameter can be SPI where x: 1, 2, or 3 to select the SPI peripheral.
  * @retval None
  */
#define __HAL_SPI_CLEAR_MODFFLAG(__HANDLE__)       \
do{                                                \
    __IO uint32_t tmpreg_modf = 0x00U;             \
    tmpreg_modf = (__HANDLE__)->Instance->SR;      \
    (__HANDLE__)->Instance->CR1 &= (~SPI_CR1_SPE); \
    UNUSED(tmpreg_modf);                           \
  } while(0U)

/** @brief  Clear the SPI OVR pending flag.
  * @param  __HANDLE__: specifies the SPI Handle.
  *         This parameter can be SPI where x: 1, 2, or 3 to select the SPI peripheral.
  * @retval None
  */
#define __HAL_SPI_CLEAR_OVRFLAG(__HANDLE__)        \
do{                                                \
    __IO uint32_t tmpreg_ovr = 0x00U;              \
    tmpreg_ovr = (__HANDLE__)->Instance->DR;       \
    tmpreg_ovr = (__HANDLE__)->Instance->SR;       \
    UNUSED(tmpreg_ovr);                            \
  } while(0U)

/** @brief  Clears the UART PE pending flag.
  * @param  __HANDLE__: specifies the UART Handle.
  *         UART Handle selects the USARTx or UARTy peripheral 
  *         (USART,UART availability and x,y values depending on device).
  */
#define __HAL_UART_CLEAR_PEFLAG(__HANDLE__)     \
  do{                                           \
    __IO uint32_t tmpreg = 0x00U;               \
    tmpreg = (__HANDLE__)->Instance->SR;        \
    tmpreg = (__HANDLE__)->Instance->DR;        \
    UNUSED(tmpreg);                             \
  } while(0U)
```

Reply:
What I meant by "discard" is to discard the value of the expression, just like discarding return value from functions explicitly.

Examples you gave look like it was done to suppress compiler warning on set but unused variables...

I'm opted for casting with `(void)`...

Reply:
I kind of prefer `UNUSED()` here (or we could define `DISCARD()` for the same cast, but that seems redundant). Reason: `UNUSED()` indicates that he read is done on purpose, and isn't just a leftover that was overlooked in an earlier change. If you are going for a cast to `(void)` then it will need a comment to achieve the same, and protect it from being removed in the next cleanup.

Reply:
I will define `DISCARD`. Reason: To me, `UNUSED` sounds like something not referenced, and `UNUSE`ing an expression is weird.

Shall we do

DISCARD instance->DR; // #define DISCARD (void)

or

DISCARD(instance->DR); // #define DISCARD(x) (void)(x)

Reply:
`DISCARD(instance->DR);` (i.e. stick to function call syntax for defines that can appear in normal code flow) is my preference.

Reply:
@ledvinap: It's clean, but it's not explicit. For a community project like this I think making code explicit where possible is a good thing.